### PR TITLE
feat: Return stuff from command handlers

### PR DESCRIPTION
**Pull request by NickSeagull**
## Description

Allows returning stuff from command handlers 

## Changes

* Added an optional `@Returns` decorator that allows specifying the return type of a command handler
* Wired the type into the GraphQL Mutation Generator
* Got result back in core so we can return it through the resolver

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly
 
## Additional Information

Here are some screenshots from tests that I've made:

Usage:
image

Return:
image

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = {
     'plugin:@typescript-eslint/recommended',
     'prettier',
     'plugin:prettier/recommended', // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
-    'plugin:import/warnings'
+    'plugin:import/warnings',
```
Comment (AdrianLorenzoDev) [inline .eslintrc.js:12]: **Minor detail**: Is this trailing comma needed?

File: .eslintrc.js [hunk 2]
```diff
@@ -45,6 +44,6 @@ module.exports = {
         allowHigherOrderFunctions: true,
       },
     ],
-    "import/no-extraneous-dependencies": ["error"]
+    'import/no-extraneous-dependencies': ['error'],
```
Comment (AdrianLorenzoDev) [inline .eslintrc.js:47]: **Minor detail**: same here about trailing comma

**Review by AdrianLorenzoDev — APPROVED** [review]
LGTM! Simple but awesome solution. Just added comments to minor details, can be ignored!

File: .eslintrc.js (hunk 1 above)
Comment (NickSeagull) [reply .eslintrc.js:12]: Yup it's an ESLint rule, I didn't add it, it was added automatically by it

**Review by juanjoman — CHANGES_REQUESTED** [review]
Love it! I'm just missing tests 🙏

File: package.json [hunk 3]
```diff
@@ -6,11 +6,14 @@
...
   "private": true,
   "scripts": {
+    "build": "lerna run compile",
+    "clean-compile": "lerna run clean && lerna clean --yes && lerna bootstrap && lerna run compile",
     "update-tocs": "npm run update-documentation-toc && npm run update-contributing-toc",
     "update-documentation-toc": "markdown-toc -i --maxdepth 4 docs/README.md",
     "update-contributing-toc": "markdown-toc -i --bullets=- --maxdepth 4 CONTRIBUTING.md"
   },
   "devDependencies": {
+    "@boostercloud/framework-provider-local": "file:packages/framework-provider-local",
```
Comment (javiertoledo) [inline package.json:16]: Why do you need to add this to `devDependencies`?

File: packages/framework-core/src/decorators/command.ts [hunk 4]
```diff
@@ -17,12 +18,49 @@ export function Command(
...
+  commandClass: Class<TCommand>,
+  methodName: string,
+  methodDescriptor: TypedPropertyDescriptor<
+    (command: TCommand, register: Register) => Promise<PrimitiveTypeOf<TReturn>>
+  >
+) => void {
+  return (commandClass) => {
+    Booster.configureCurrentEnv((config): void => {
+      if (config.commandHandlerReturnTypes[commandClass.name]) {
+        throw new Error(`A command handler for the command ${commandClass.name} is already registered`)
```
Comment (javiertoledo) [inline packages/framework-core/src/decorators/command.ts:52]:
```suggestion
        throw new Error(`A command handler return type for the command ${commandClass.name} is already registered`)
```

**Review by javiertoledo — APPROVED** [review]
It's a pity that there's not a way to get the return type automatically here. I wonder if it would be worth redesigning the handler as an instance method to be able to use `Reflect.getMetadata`...

File: package.json (hunk 3 above)
Comment (NickSeagull) [reply package.json:16]: Thanks, this was a test

**Comment by NickSeagull** [comment]
Will add tests later

**Pull request merged**